stock_picking_invoice_link/init_hook.py

```diff
@@ -0,0 +1,92 @@
+# Copyright 2023 ForgeFlow, S.L.
+# Copyright 2024 Tecnativa - David Vidal
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+import logging
+from collections import deque
+
+from odoo import SUPERUSER_ID, api
+from odoo.tools import sql
+
+_logger = logging.getLogger(__name__)
+
+
+def _create_picking_count(cr):
+    """Extracted to reuse in migration script. TODO: It can be merged into a sole
+    method in v16"""
+    _logger.info("Creating column: picking_count for table account_move")
+    cr.execute(
+        """
+    ALTER TABLE account_move
+        ADD COLUMN IF NOT EXISTS picking_count INTEGER;
+    """
+    )
+
+
+def _populate_picking_count(cr):
+    """Extracted to reuse in migration script. TODO: It can be merged into a sole
+    method in v16"""
+    _logger.info("Populating values for column picking_count in table account_move")
+    cr.execute(
+        """
+        UPDATE account_move am SET picking_count = t.picking_count
+        FROM (
+            SELECT
+                am.id AS account_move_id,
+                COUNT(DISTINCT sm.picking_id) AS picking_count
+            FROM
+                account_move am
+                JOIN account_move_line aml ON aml.move_id = am.id
+                JOIN stock_move_invoice_line_rel rel
```

Comment:
Why not using here account_move_stock_picking_rel? Maybe you just need to swap order